src/main/java/de/hpi/datastreams/processors/ClassifierProcessor.java

```diff
@@ -0,0 +1,39 @@
+package de.hpi.datastreams.processors;
+
+import de.hpi.datastreams.apps.App;
+import de.hpi.datastreams.messages.DataMessage;
+import de.hpi.datastreams.producer.ProducerBuilder;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+import java.util.UUID;
+
+import static de.hpi.datastreams.apps.App.CLASSIFICATION_OUTPUT_TOPIC;
+
+public class ClassifierProcessor extends AbstractProcessor<Long, DataMessage> {
+
+    private KeyValueStore<Integer, Float> weights;
+    private Producer<Long, Float> classificationMessageProducer;
+
+    @Override
+    public void init(ProcessorContext context) {
+        super.init(context);
+        this.weights = (KeyValueStore<Integer, Float>) context.getStateStore(App.WEIGHTS_STORE);
+        this.classificationMessageProducer = ProducerBuilder.buildFloatValue("classificationProducer-" + UUID.randomUUID().toString());
+    }
+
+    @Override
+    public void process(Long key, DataMessage value) {
+        // TODO: calculate the prediction
+        float prediction = 0;
+
+        System.out.println("ClassifierProcessor - Calculating prediction for message: " + value.getVectorClock());
+        System.out.println("ClassifierProcessor - Size of weights: " + this.weights.approximateNumEntries());
```

Comment:
Do you think it makes sense to rethink the partitioning of the weights state store?
Right now we kind of utilize the `partitionKey` for our own purposes (the weight id).